### PR TITLE
Add release pipeline

### PR DESCRIPTION
**Pull request by aneuwald-ctw**
**User-Facing Changes**
N/A

**Description**
This PR introduces a new Release pipeline to automate the release process, which includes:
- Building binaries for Windows, Linux, and macOS.
- Automatically updating the minor version.
- Creating a tag with the updated version.
- Publishing a release with the built binaries.

Additionally, the post-release workflow has been updated to:
- Remove the version bump step, as it's now handled in the release workflow.
- Exclude the build for Linux/ARM64 to reduce compile time.
- Tag the latest Docker image with both latest and the current version in the format x.x.x.

This PR also fixes a warning in the Dockerfile, ensuring the AS keyword is capitalized: ```FROM node:16 AS build```

**Checklist**

- [x] The web version was tested and it is running ok
- [x] The desktop version was tested and it is running ok
- [ ] I've updated/created the storybook file(s)

### CODE REVIEW

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -68,47 +32,41 @@ jobs:
 
   docker:
     name: Publish to GHCR
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (laisspportugal) [inline .github/workflows/post-release.yml:35]: Same question

File: .github/workflows/post-release.yml [hunk 2]
```diff
@@ -68,47 +32,41 @@ jobs:
 
   docker:
     name: Publish to GHCR
-    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (aneuwald-ctw) [reply .github/workflows/post-release.yml:71]: We can normalize to use 20.04 in all places. In some places we have latest in others 20.04, so I chose to use that one. But I can normalize to use 20.04

File: .github/workflows/post-release.yml [hunk 3]
```diff
@@ -68,47 +32,41 @@ jobs:
...
-        uses: docker/setup-buildx-action@v3
+      - name: Format version
+        id: format_version
+        run: |
+          VERSION=${{ github.event.release.tag_name }}
+          FORMATTED_VERSION=$(echo "$VERSION" | sed 's/^v//')
+          echo "::set-output name=formatted_version::$FORMATTED_VERSION"
 
-      - name: Build and push
+      - name: Build and Push Docker image
```
Comment (laisspportugal) [inline .github/workflows/post-release.yml:65]: I would recommend to keep it one of those both ways

- Build and Push Docker Image
- Build and puh Docker image

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        working-directory: packages/suite
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Build prod files
+        run: |
+          yarn run desktop:build:prod
+
+      - name: Build software in serie
```
Comment (laisspportugal) [inline .github/workflows/release.yml:41]: Suggestion:

Build software for multiple OS
In order to make it more understandable

**Review by laisspportugal — COMMENTED** [review]
Just comments with questions and optional changes. Besides them, looks good to me.

**Review by laisspportugal — APPROVED** [review]
Looks good to me

**Pull request merged**